### PR TITLE
python: upgrade setuptools

### PR DESCRIPTION
**Pull request by tdsmith**
New setuptools version stopped vendoring their dependencies so the setup process got a little more involved. Bootstrap with ensurepip and then install them all with the (probably old) pip and setuptools installed with the interpreter.

### CODE REVIEW

**Comment by tdsmith** [comment]
postinstall failed but the error isn't displayed, whomp

**Comment by ilovezfs** [comment]
It's reproducible for me locally as well. The error is
```
Warning: The post-install step did not complete successfully
You can try again using `brew postinstall python`
==> No such file or directory - /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/distutils.cfg
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/python.rb:255:in `unlink'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/python.rb:255:in `unlink'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/python.rb:255:in `post_install'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1029:in `block in run_post_install'
/usr/local/Homebrew/Library/Homebrew/formula.rb:887:in `with_logging'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1028:in `run_post_install'
/usr/local/Homebrew/Library/Homebrew/postinstall.rb:16:in `<main>'
```

File: Formula/python.rb [hunk 1]
```diff
@@ -243,21 +266,11 @@ def post_install
...
-    (libexec/"setuptools").cd { system "#{bin}/python", *setup_args }
-    (libexec/"pip").cd { system "#{bin}/python", *setup_args }
-    (libexec/"wheel").cd { system "#{bin}/python", *setup_args }
-
-    # When building from source, these symlinks will not exist, since
-    # post_install happens after linking.
-    %w[pip pip2 pip2.7 easy_install easy_install-2.7 wheel].each do |e|
-      (HOMEBREW_PREFIX/"bin").install_symlink bin/e
+    system bin/"python", "-sS", "-m", "ensurepip"
+    libexec.children.each do |path|
```
Comment (ilovezfs) [inline Formula/python.rb]: .each_child maybe

**Comment by ilovezfs** [comment]
@tdsmith any thoughts on whether to revision bump here?

**Comment by tdsmith** [comment]
I don't think there are any important behavior changes in setuptools so I'd be inclined not to bump.

**Comment by ilovezfs** [comment]
OK :)

**Comment by tdsmith** [comment]
setuptools revendored its dependencies so this is no longer necessary. https://setuptools.readthedocs.io/en/latest/history.html#v36-0-0

**Pull request closed without merging**